### DISCUSSION TITLE
asset naming for 2-extension, hashed/compressed assets -- e.g. 'file.hash.js.br' ?

### DISCUSSION THREAD

**Discussion by pgnd** [webdeveric/webpack-assets-manifest#204] [Q&A]
i'm adding static-compressed assets to my usual webpack5 builds.  e.g., using brotli compression,

```
ls -al public/build/js/bootstrap.afcb5d.js*
	-rw-rw-r--+ 1 wwwrun www 57K Dec 16 17:33 public/build/js/bootstrap.afcb5d.js
	-rw-rw-r--+ 1 wwwrun www 14K Dec 16 17:33 public/build/js/bootstrap.afcb5d.js.br
```

with my current manifest config,
```
const assetsManifest = () => ({
	plugins: [
		new WebpackAssetsManifest({
			enabled: true,
			contextRelativeKeys: false,
			entrypoints: false,
			entrypointsKey: false,
			entrypointsUseAssets: false,
			fileExtRegex: /\.\w{2,4}\.(?:map|gz|br)$|\.\w+$/i,
			integrity: true,
			integrityHashes: ['sha256', 'sha384', 'sha512'],
			merge: false,
			output: assets-manifest.json,
			publicPath(filename, manifest) {
				switch (manifest.getExtension(filename).substr(1).toLowerCase()) {
					case 'jpg':
					case 'jpeg':
					case 'gif':
					case 'png':
					case 'svg':
					case 'ico':
						return `/build/${filename}`;
					case 'css':
						return `/build/${filename}`;
					case 'js':
						return `/build/${filename}`;
					default:
						return `/build/${filename}`;
				}
			},
			sortManifest: true,
			customize: (o) => {
				if (o.key.startsWith('img/')) {
					return { key: o.key.split('img/')[1], value: o.value };
				}
				return o;
			},
		}),
	],
});
```

the generated manifest includes,
```
grep bootstrap  public/build/assets-manifest.json
  "bootstrap.js": {
    "src": "/build/js/bootstrap.afcb5d.js",
  "js/bootstrap.afcb5d.js.br": {
    "src": "/build/js/bootstrap.afcb5d.js.br",
```

I want the '.br' in the same unhashed key asset form/path as the *UN*compressed asset.

So, e.g., expecting in this case, instead
```
grep bootstrap  public/build/assets-manifest.json
  "bootstrap.js": {
    "src": "/build/js/bootstrap.afcb5d.js",
  "bootstrap.js.br": {
    "src": "/build/js/bootstrap.afcb5d.js.br",
```

I thought the mod of
```
fileExtRegex: /\.\w{2,4}\.(?:map|gz|br)$|\.\w+$/i,
```

would be sufficient.  Apparently not.

Does `w-a-m` have native support for handling `js.br` type 2-extensions asset naming, specifically for compressed assets (.br, .gz, etc)?

Or a good example of some name parsing conditionals that'll make it behave as intended?

**Comment by webdeveric** [comment]
@pgnd Are you using [compression-webpack-plugin](https://webpack.js.org/plugins/compression-webpack-plugin/#using-brotli ) to generate `.br` files? If so, have you configured the [`filename` option](https://webpack.js.org/plugins/compression-webpack-plugin/#filename)?

You can also try the [`transform`](https://github.com/webdeveric/webpack-assets-manifest#transform) and [`customize`](https://github.com/webdeveric/webpack-assets-manifest#customize) hooks to modify the manifest contents to be whatever you need.
